torch/optim/adam.py

```diff
@@ -328,7 +338,7 @@ def _multi_tensor_adam(params: List[Tensor],
                        weight_decay: float,
                        eps: float,
                        maximize: bool,
-                       capturable: bool):
+                       differentiable: bool):
```

Comment:
The foreach ops don't support autograd. So we should have an assert here that differentiable is False (same for sgd btw).
This can be done in a separate PR though.

Reply:
Added the assert to this one, will fix SGD too in the next PR :)